### PR TITLE
feat(Quiz): add quiz component

### PR DESCRIPTION
**Pull request by anderson-olive**
# Description

This PR add the quiz page & components to the project, I've also added the mobile layout version.

## Changes Included:
- Fix mobile layout & add new hamburger navigation
- Add Quiz Page
- Add Quiz Result Page
- Add "QuizItem" Component and "Error" & "Success" variants
- Add "services/quiz" to support retrieve the quiz questions
- UI Improvements

## Demo

Desktop:

https://github.com/nandoramos/youtube-gpt/assets/106548895/64ecbefd-f345-4f61-9cde-9e356b5e10bb

Mobile:

https://github.com/nandoramos/youtube-gpt/assets/106548895/75cd5371-21cc-41ed-93f0-4686a559f4b8

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: youtube-gpt-client/src/components/quiz/index.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  const selectAnswer = (questionIndex: number, correctAnswer: boolean) => {
+    if (!canSelect(questionIndex)) return;
+
+    const response = [
+      ...(quizStatus || []),
+      { questionIndex: questionIndex, response: correctAnswer },
+    ];
+
+    setQuizStatus(response);
+    moveToNextQuestion(questionIndex, response);
```
Comment (nandoramos) [inline youtube-gpt-client/src/components/quiz/index.tsx]: Here we shuld add a timeout in order to see the results

**Review by nandoramos — APPROVED** [review]
👏 Very nice!

File: youtube-gpt-client/src/components/header/index.tsx [hunk 2]
```diff
@@ -1,27 +1,101 @@
...
-        <Link href="/team">{t('team')}</Link>
-        <LanguageSwitcher />
+    <>
+      <Flex
+        justifyContent="space-between"
+        alignItems="center"
+        padding="15px 50px"
+        color="#fff"
+        height="60px"
+        display={['none', 'none', 'flex', 'flex']}
```
Comment (nnicolopez) [inline youtube-gpt-client/src/components/header/index.tsx]: Don't get what is this line for. The Chakra-UI Flex component has already `display: flex` as default.

File: youtube-gpt-client/src/components/header/index.tsx [hunk 3]
```diff
@@ -1,27 +1,101 @@
...
       </Flex>
-    </Flex>
+      <Flex
+        justifyContent="space-between"
+        alignItems="center"
+        padding="15px 0px 15px 10px;"
+        color="#fff"
+        height="60px"
+        width="100%"
+        display={['flex', 'flex', 'none', 'none']}
```
Comment (nnicolopez) [inline youtube-gpt-client/src/components/header/index.tsx]: Don't get what is this line for. The Chakra-UI Flex component has already display: flex as default.
Same as above.

Comment (anderson-olive) [reply youtube-gpt-client/src/components/header/index.tsx]: Im using [responsive styles](https://chakra-ui.com/docs/styled-system/responsive-styles), using that, we can set the media query property for each media query, and in this case, I want to show it on mobile but not on desktop (smaller screens). I'll simplify this to only 2 media queries (mobile and desktop)

File: youtube-gpt-client/src/components/mainContent/index.tsx [hunk 4]
```diff
@@ -9,7 +9,7 @@ const MainContent = ({ children }: PropsWithChildren) => {
       backgroundColor="#fff"
       borderRadius="20px 20px 0 0"
       minHeight="calc(100vh - 60px)"
-      padding="55px 50px"
+      padding={['20px 10px', '55px 50px']}
```
Comment (nnicolopez) [inline youtube-gpt-client/src/components/mainContent/index.tsx:12]:
```suggestion
      padding="20px 10px 55px 50px"
```

File: youtube-gpt-client/src/components/header/index.tsx (hunk 2 above)
Comment (anderson-olive) [reply youtube-gpt-client/src/components/header/index.tsx]: I don't want to show this component on mobile devices, so we are using the [responsive styles](https://chakra-ui.com/docs/styled-system/responsive-styles) to hide it

File: youtube-gpt-client/src/components/mainContent/index.tsx (hunk 4 above)
Comment (anderson-olive) [reply youtube-gpt-client/src/components/mainContent/index.tsx:12]: Same here, we are using different values to each device 20px and 10px to mobile and 55px and 50px to desktop

**Review by nnicolopez — APPROVED** [review]

File: youtube-gpt-client/src/components/header/index.tsx (hunk 3 above)
Comment (nnicolopez) [reply youtube-gpt-client/src/components/header/index.tsx]: didn't know that feature, nice!

**Pull request merged**